src/rpc/RPCHelpers.cpp

```diff
@@ -1421,6 +1421,12 @@ parseBook(boost::json::object const& request)
             "Invalid field 'taker_pays.issuer', expected non-XRP "
             "issuer."};
 
+    if (isXRP(pay_currency) && isXRP(pay_issuer))
+        return Status{
+            Error::rpcSRC_CUR_MALFORMED,
+            "Unneeded field 'taker_pays.issuer' for XRP currency "
```

Comment:
Is this the same error you would get from rippled? LGTM if it is.

Reply:
@godexsoft this was a reused error message for a new conditional to handle the new case of XRP pay currency with an XRP specified issuer (with the latter not being necessary at all). This solves one of the cases @ManiMounikaKunasani added later